src/Services/Settings.js

```diff
@@ -57,11 +58,43 @@ class Settings {
 
     // Ensure new settings variables get added by merging
     // all the settings, where the current values take precedence. 
-    var currentSettings = settings.getAll();
+    var currentSettings = deepMergeSettings(initialSettings, settings.getAll());
```

Comment:
What was the impetus for this change? Seems out of scope.

During development, there were a lot of settings issues caused by extra merging logic here (you can see my comments don't actually reflect reality, since no merging took place except on top level elements). I worry this extra logic will bring those issues back, and we'll need extra testing to ensure they're not there. 

If there's no immediate need for this extra merging I'd vote to take it out until we need it.

If we *do* need this change, you should likely replace your `deepMergeSettings` function with the `.merge` function provided by `lodash`.

Reply:
Capturing some OOB conversation here:

Unfortunately this change doesn't work without this. If you revert line 61 and then try to run with a settings file which doesn't have the `vmErrorsOnRPCResponse` field in the `initialSettings.server` object, it'll wind up starting the chain without specifying the `vmErrorsOnRPCResponse` explicitly. This will give you the desired default behavior (`true`) because that's the default setting that `ganache-core` has, but the settings UI will show the opposite of what's actually running in the chain (`false`).